### DISCUSSION TITLE
[Bug] 插件无法正常调用，回复需要等待然后不再回答

### DISCUSSION THREAD

**Discussion by angermanger** [lobehub/lobehub#2368] [General Question] [answered]
### 💻 系统环境

Windows

### 📦 部署环境

Official Preview

### 🌐 浏览器

Chrome

### 🐛 问题描述

我想使用askforpdf插件，但在启用后仍然无法调用该插件，我的api是sourcegraph的api，使用模型是gpt4，上传的PDF url是用catbox.moe生成的
![image](https://github.com/lobehub/lobe-chat/assets/168886893/ec71f4aa-5a81-4a02-b6b1-497046a51296)
![image](https://github.com/lobehub/lobe-chat/assets/168886893/204890e5-a98e-4af6-999a-64e6189fd0a2)

[a comment by an automated account is not shown]

**Comment by angermanger** [comment]
https://files.catbox.moe/7rbd2i.pdf这是pdf的url

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
你的供应商提供的gpt4 可能不是官方原版，以至于不支持function calling

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
![image](https://github.com/lobehub/lobe-chat/assets/28616219/26943629-4e11-4707-b1dc-200822ee1dfe)

**Comment by angermanger** [comment]
这个是sourcegraph这个平台推出的api，集合了很多模型，Claude和gpt都有，可能是因为反代接口和上游不太适配，导致没办法读取？

[a comment by an automated account is not shown]

**Comment by angermanger** [comment]
curl \
  -H 'Authorization: token sgp_a0d7ccb4f752ea73_6114fcfcf4a56a9cf00deb5d6719a7dca93915a5' \
  -d '{"query":"query { currentUser { username } }"}' \
  https://sourcegraph.com/.api/graphql
他这个原来官方的api参考格式是这样的，但我填上去也连接不了

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
> 这个是sourcegraph这个平台推出的api，集合了很多模型，Claude和 gpt 都有，可能是因为反代接口和上游不太适配，导致没办法读取？

不是，大概率是你用的这个服务商根本就没有用官方的 API ，导致不支持 Function Calling，自然不能用插件。我们之前遇到过好几个用户用了无良服务商的逆向 GPT 4，都没法用 Function Calling 的。

你可以试试我们合作的一家 API 服务商，他们的是官方原版的 API ，是能正常使用的： https://lobe.li/XHnZIUP

[a comment by an automated account is not shown]
